### ISSUE TITLE
Student can register for a course with invalid GitHub URL

### ISSUE THREAD

**Issue by gotonode** [UniversityOfHelsinkiCS/labtool#1022]
GitHub repository format is `https://github.com/user/repo`, but currently the user can enter anything in the related field. Should this be checked before the data is saved?

**Comment by mjaakko** [comment]
Currently a warning is shown if the GitHub repo does not exist (this should also happen if the URL is invalid e.g. `https://github.com/abcfondfoosadf`). We also don't want to disallow using other git services and it would be practically impossible to validate URL formats of all possible git services.

**Comment by hisahi** [comment]
Might be worth implementing for GitHub only but that is about it

**Comment by hisahi** [comment]
I will take a look at whether it's possible to check GitHub URLs

**Comment by hisahi** [comment]
The existing code already verifies invalid repositories, marking them as nonexistent. This logic should be sufficient, as the teacher can already tell that the student used an invalid URL. Therefore I'm choosing to close this, as nothing is really needed, and we cannot validate URLs for all possible alternative services.
